app/controllers/owner/products_controller.rb

```diff
@@ -0,0 +1,57 @@
+module Owner
+  class ProductsController < OwnerController
+    before_action :set_product, only: %i[show edit update destroy]
+    before_action :set_salon, only: %i[new create]
+
+    def new
+      @product = Product.new
+    end
+
+    def edit; end
+
+    def create
+      @product = Product.new(product_params)
+      @product.salon = @salon
+
+      respond_to do |format|
+        if @product.save
+          format.html { redirect_to product_url(@product), notice: "Product was successfully created." }
+        else
+          format.html { render :new, status: :unprocessable_entity }
+        end
+      end
+    end
+
+    def update
+      respond_to do |format|
+        if @product.update(product_params)
```

Comment:
With such logic you will be able to update any product you want. I guess, you need to be able update only products related to the salon. 
If I will play a little bit with your UI, I will be able to do a lot of harm to your database :)

Here should be something like `@salon.products.find(params[:id])`
In this case, user will be able to edit product related to this salon only.